File: OOP_var/1.rb
class Cal
    def initialize(v1, v2)
        @val1, @val2 = v1, v2
    end

    def add
        return @val1 + @val2
    end

    def subtract
        return @val1 - @val2
    end

    def getV1
        return @val1
    end

    def setV1(v1)
        if v1.is_a?(Integer)
            @val1 = v1
        end
    end
end

cal1 = Cal.new(10, 10)
p cal1.add
p cal1.subtract

cal1.setV1("one")
p cal1.getV1

File: OOP_var/1.py
class Cal(object):
    def __init__(self, v1, v2):
        self.v1 = v1
        self.v2 = v2
    
    def add(self):
        return self.v1 + self.v2

    def subtract(self):
        return self.v1 - self.v2

    def getV1(self):
        return self.v1

    def setV1(self, v1):
        if isinstance(v1, int):
            self.v1 = v1

cal1 = Cal(10, 10)
print(cal1.add())
print(cal1.subtract())

cal1.setV1("one")
print(cal1.getV1())